### PR TITLE
002 job spec each role

### PR DESCRIPTION
**Pull request by dcorry586**

### CODE REVIEW

File: services/JobService.ts [hunk 1]
```diff
@@ -5,9 +5,19 @@ export default class JobService {
     async getJobs(): Promise<Job[]> {
         try{
             const response = await axiosInstance.get('/api/jobs');
+            // console.log("jobs --> ", response.data);
```
Comment (TaylorJCrawford) [inline services/JobService.ts]: Needs Removed.

File: models/JobReset.ts
Comment (dcorry586) [inline models/JobReset.ts:1]: remove

File: test/unit/JobServiceTest.ts [hunk 2]
```diff
@@ -20,21 +20,42 @@ describe('JobService', function () {
...
         try {
             await jobService.getJobs();
         } catch (e) {
             error = e;
         }
         expect(String(error)).to.equal('Error: Could not get jobs');
     });
+
+    it('Get job by id should return a job from the API', async () => {
+        const id: number = 1; // Define your id here
```
Comment (dcorry586) [inline test/unit/JobServiceTest.ts]: remove comment

File: services/JobService.ts [hunk 3]
```diff
@@ -11,4 +11,13 @@ export default class JobService {
             throw new Error('Could not get jobs');
         }
     }
+
+async getJobById(id: number): Promise<Job> {
+        try{
+            const response = await axiosInstance.get('/api/jobs/' + id);
```
Comment (dcorry586) [inline services/JobService.ts]: remove concatenation - insert template literals

File: test/unit/JobServiceTest.ts [hunk 4]
```diff
@@ -20,21 +20,42 @@ describe('JobService', function () {
...
+        mock.onGet(`/api/jobs/${id}`).reply(200, job);
+
+        let result: Job = await jobService.getJobById(id);
+
+        expect(result).to.deep.equal(job);
+    });
+
+    it('should throw an error when the API returns a non-200 status', async () => {
+        const id: number = 111; // Define your id here
```
Comment (dcorry586) [inline test/unit/JobServiceTest.ts]: remove comment

File: views/JobTitles.ejs [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+</head>
+
+<body>
+
+</body>
+
+</html>
+
+<body>
+  <nav class="navbar navbar-expand-lg bg-body-tertiary">
```
Comment (ghost) [inline views/JobTitles.ejs]: The navbar shouldn't be in it

Comment (ghost) [reply views/JobTitles.ejs]: Actually remove whole file

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: controllers/JobsController.ts [hunk 6]
```diff
@@ -1,17 +1,27 @@
...
+    });
+    app.get('/jobs/:id', async (req: Request, res: Response) => {
+        const id = req.params.id;
+        let job: Job;
+        try {
+            job = await jobService.getJobById(Number(id));
+        } catch (e) {
+            console.error(e);
+        }
+        res.render('pages/JobById', { job, pageTitle: 'Job' });
```
Comment (glenn-hamilton) [inline controllers/JobsController.ts]: JobById might not be a good name for this template

File: public/style.css [hunk 7]
```diff
@@ -53,7 +53,8 @@ h2 {
     font-family: 'Century Gothic', sans-serif;
     color: #262E4D;
     font-size: 2.2em;
-    margin-bottom: 30px;
+    margin-top: 30px;
+    margin-bottom: 20px;
```
Comment (glenn-hamilton) [inline public/style.css]: Could this have unintended consequences elsewhere?

**Review by glenn-hamilton — APPROVED** [review]

**Pull request merged**